src/Illuminate/Queue/Queue.php

```diff
@@ -178,6 +178,27 @@ protected function getDisplayName($job)
                         ? $job->displayName() : get_class($job);
     }
 
+     /**
+     * Get the maximum number of attempts for an object-based queue handler.
+     *
+     * @param  mixed  $job
+     * @return mixed
+     */
+    public function getJobTries($job)
+    {
+        if (! method_exists($job, 'tries') && ! isset($job->tries)) {
+            return;
+        }
+
+        if (isset($job->tries)) {
+            return $job->tries;
+        }
+
+        if (method_exists($job, 'tries') && ! is_null($job->tries())) {
+            return $job->tries();
```

Comment:
`tries()` has been called twice by now. Maybe this is a problem when the function performs some advanced logic?